File: baekjoon/shim/Java/단계별로_풀어보기/기하/n_2477.java
package Study_2022.baekjoon.shim.Java.단계별로_풀어보기.기하;

// import java.io.*;

// public class n_2477 {
//     public static void main(String[] args)throws IOException{
//         BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
//         int n=Integer.parseInt(br.readLine()), arr[]=new int[6];
//         int w=0, h=0, w2=0, h2=0;
//         for(int i=0; i<6; i++){
//             String st[]=br.readLine().split(" ");
//             arr[i]=Integer.parseInt(st[1]);
//             if(i%2==0)
//                 w=Math.max(w,arr[i]);
//             else
//                 h=Math.max(h,arr[i]);
//         }
//         for(int i=0; i<6; i++){
//             if(i%2==0 && h==arr[(i+5)%6]+arr[(i+1)%6])
//                 w2=arr[i];
//             else if(i%2==1 && w==arr[(i+5)%6]+arr[(i+1)%6])
//                 h2=arr[i];
//         }
//         System.out.println(n*(w*h-w2*h2));
//     }
// }

//리벤지전
import java.io.*;

public class n_2477 {
    public static void main(String[] args)throws IOException{
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        int n=Integer.parseInt(br.readLine()), arr[]=new int[6], max_h=0, max_w=0, h=0, w=0;
        for(int i=0; i<6; i++){
            String st[]=br.readLine().split(" ");
            arr[i]=Integer.parseInt(st[1]);
            if(i%2==0)
                max_h=Math.max(max_h, arr[i]);
            else
                max_w=Math.max(max_w, arr[i]);
        }
        for(int i=0; i<6; i++){
            if(i%2==0&&max_h==(arr[i%6]+arr[(i+2)%6]))
                h=arr[(i+1)%6];
            else if(i%2==1&max_w==(arr[i%6]+arr[(i+2)%6]))
                w=arr[(i+1)%6];
        }
        System.out.println((max_h*max_w-h*w)*n);
    }
}

File: baekjoon/shim/Python/단계별로_풀어보기/기하/n_2477.py
# import sys
# n = int(sys.stdin.readline())
# l = []
# w = h = w2 = h2 = 0
# for i in range(6):
#     _,m = map(int, sys.stdin.readline().split())
#     l.append(m)
#     if i % 2 == 0:
#         w = max(w, m)
#     else:
#         h = max(h, m)
# for i in range(6):
#     if i % 2 == 0 and h == l[(i+5) % 6]+l[(i+1) % 6]:
#         w2 = l[i]
#     elif i % 2 == 1 and w == l[(i+5) % 6]+l[(i+1) % 6]:
#         h2 = l[i]
# print(n*(w*h - w2*h2))

#리벤지전
import sys
input=sys.stdin.readline
n=int(input())
max_h=max_w=h=w=0
arr=[]
for i in range(6):
    _,m=map(int,input().split())
    arr.append(m)
    if i%2==0:
        max_h=max(max_h,m)
    else:
        max_w=max(max_w, m)
for i in range(6):
    if i%2==0 and max_h==(arr[i%6]+arr[(i+2)%6]):
        h=arr[(i+1)%6]
    elif i%2==1 and max_w==(arr[i%6]+arr[(i+2)%6]):
        w=arr[(i+1)%6]
print((max_h*max_w-w*h)*n)